hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/grid/AbstractGrid.java

```diff
@@ -52,56 +82,595 @@ protected AbstractGrid(final VaadinMessageSource i18n, final UIEventBus eventBus
+         */
+        protected GeneratedPropertyContainer decorate(final T container) {
+            return new GeneratedPropertyContainer(container);
+        }
+    }
+
+    /**
+     * Support for single selection on the grid.
+     */
+    protected class SingleSelectionSupport {
+        private boolean enabled;
+
+        public SingleSelectionSupport(final boolean enabled) {
+            if (enabled) {
+                enable();
+            } else {
+                disable();
+            }
+        }
+
+        public void enable() {
+            enabled = true;
+            setSelectionMode(SelectionMode.SINGLE);
+        }
+
+        public void disable() {
+            enabled = false;
+            setSelectionMode(SelectionMode.NONE);
+        }
+
+        /**
+         * Selects the first row if available and enabled.
+         */
+        public void selectFirstRow() {
+            if (!enabled) {
+                return;
+            }
+
+            final Indexed container = getContainerDataSource();
+            final int size = container.size();
+            if (size > 0) {
+                refreshRows(getContainerDataSource().firstItemId());
+                getSingleSelectionModel().select(getContainerDataSource().firstItemId());
+            } else {
+                getSingleSelectionModel().select(null);
+            }
+        }
+
+        /**
+         * Clears the selection.
+         */
+        public void clearSelection() {
+            getSingleSelectionModel().select(null);
+        }
+
+        /**
+         * Offers type-save access to the single selection model.
+         *
+         * @return SelectionModel.Single of the grid.
+         */
+        protected SelectionModel.Single getSingleSelectionModel() {
+            return (SelectionModel.Single) getSelectionModel();
+        }
+    }
+
+    /**
+     * CellStyleGenerator that concerns about alignment in the grid cells.
+     */
+    protected static class AlignCellStyleGenerator implements CellStyleGenerator {
+        private static final long serialVersionUID = 5573570647129792429L;
+
+        private final String[] left;
+        private final String[] center;
+        private final String[] right;
+
+        /**
+         * Constructor.
+         *
+         * @param left
+         *            list of propertyIds that should be left-aligned
+         * @param center
+         *            list of propertyIds that should be center-aligned
+         * @param right
+         *            list of propertyIds that should be right-aligned
+         */
+        public AlignCellStyleGenerator(final String[] left, final String[] center, final String[] right) {
+            this.left = left;
+            this.center = center;
+            this.right = right;
+        }
+
+        @Override
+        public String getStyle(final CellReference cellReference) {
+            if (center != null && Arrays.asList(center).contains(cellReference.getPropertyId())) {
```

Comment:
It's very inefficient using a collection here. Streams or `ArrayUtils.contains` is the better way

http://www.programcreek.com/2014/04/check-if-array-contains-a-value-java/